### ISSUE TITLE
Replace runtime a11y errors with discriminated unions

### ISSUE THREAD

**Issue by ynotdraw** [lifeomic/chroma-react#118]
Some of our form components throw runtime errors in dev mode.  The most common case is when a `label` is falsy, we want to require `aria-label` for a11y reasons.  Instead of doing a runtime error, we should be able to leverage discriminated unions https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions

**Comment by ynotdraw** [comment]
This is a bit tricky to do properly at the moment.  We will stick with runtime errors for now.
